File: TarsEngine.FSharp.Core/TreeOfThought/IThoughtNode.fs
namespace TarsEngine.FSharp.Core.TreeOfThought

open System.Collections.Generic

/// <summary>
/// Interface for a thought node in a tree-of-thought.
/// </summary>
type IThoughtNode =
    /// <summary>
    /// Gets the thought content.
    /// </summary>
    abstract member Thought : string
    
    /// <summary>
    /// Gets the child nodes.
    /// </summary>
    abstract member Children : IReadOnlyList<IThoughtNode>
    
    /// <summary>
    /// Gets the evaluation metrics.
    /// </summary>
    abstract member Evaluation : EvaluationMetrics option
    
    /// <summary>
    /// Gets a value indicating whether the node has been pruned.
    /// </summary>
    abstract member Pruned : bool
    
    /// <summary>
    /// Gets the metadata.
    /// </summary>
    abstract member Metadata : IReadOnlyDictionary<string, obj>
    
    /// <summary>
    /// Gets the score of the node.
    /// </summary>
    abstract member Score : float

File: Legacy_CSharp_Projects/TarsEngine.Services.Abstractions/TreeOfThought/IThoughtNode.cs
using System.Collections.Generic;

namespace TarsEngine.Services.Abstractions.TreeOfThought
{
    /// <summary>
    /// Interface for a node in a thought tree.
    /// </summary>
    public interface IThoughtNode
    {
        /// <summary>
        /// Gets the thought content.
        /// </summary>
        string Thought { get; }

        /// <summary>
        /// Gets the child nodes.
        /// </summary>
        IReadOnlyList<IThoughtNode> Children { get; }

        /// <summary>
        /// Gets the evaluation metrics.
        /// </summary>
        EvaluationMetrics? Evaluation { get; }

        /// <summary>
        /// Gets a value indicating whether the node has been pruned.
        /// </summary>
        bool Pruned { get; }

        /// <summary>
        /// Gets the metadata.
        /// </summary>
        IReadOnlyDictionary<string, object> Metadata { get; }

        /// <summary>
        /// Gets the score of the node.
        /// </summary>
        double Score { get; }
    }
}
